editor/editor_config.cpp

```diff
@@ -120,9 +120,22 @@ EditorConfig::EditorConfig(string const & fileName)
   Reload();
 }
 
-bool EditorConfig::GetTypeDescription(vector<string> const & classificatorTypes,
+bool EditorConfig::GetTypeDescription(vector<string> && classificatorTypes,
```

Comment:
Не может тип повторяться. Не понимаю, зачем разносить логически связанные вещи.

> On 4.4.2016, at 18.28, ygorshenin <notifications@github.com> wrote:
> 
> In editor/editor_config.cpp:
> 
> > @@ -120,9 +120,22 @@ EditorConfig::EditorConfig(string const & fileName)
> >    Reload();
> >  }
> >  
> > -bool EditorConfig::GetTypeDescription(vector<string> const & classificatorTypes,
> > +bool EditorConfig::GetTypeDescription(vector<string> && classificatorTypes,
> Так разнеси это по отдельным методам - пусть один метод удаляет тип building из classificatorTypes, второй метод уже заполняет outDesc. Так у тебя логика перемешана - с одной стороны ты удаляешь ненужные типы с другой что-то запоняешь. Может ли такое быть что "building" встречается несколько раз в classificatorTypes?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>